pkg/auth/handler/webapp/verify_identity.go

```diff
@@ -166,36 +263,50 @@ func (h *VerifyIdentityHandler) ServeHTTP(w http.ResponseWriter, r *http.Request
 		return nil
 	})
 
-	ctrl.PostAction("submit", func() error {
-		inputFn := func() (input interface{}, err error) {
-			err = VerifyIdentitySchema.PartValidator(VerifyIdentityRequestSchema).ValidateValue(FormToJSON(r.Form))
-			if err != nil {
-				return
-			}
+	ctrl.PostAction(VerifyIdentityActionSubmit, func() error {
+		result, err := ctrl.InteractionPost(inputFn)
+		if err != nil {
+			return err
+		}
+		result.WriteResponse(w, r)
+		return nil
+	})
 
-			code := r.Form.Get("x_code")
+	ctrl.PostAction(VerifyIdentityActionResume, func() error {
+		result, err := ctrl.EntryPointPost(opts, intent, inputFn)
+		if err != nil {
+			return err
+		}
+		result.WriteResponse(w, r)
+		return nil
+	})
 
-			input = &InputVerificationCode{
-				Code: code,
-			}
-			return
+	ctrl.PostAction(VerifyIdentityActionUpdateSessionStep, func() error {
+		code, err := h.Verifications.GetCode(verificationCodeID)
+		if err != nil {
+			return err
 		}
 
-		session, err := ctrl.InteractionSession()
-		if errors.Is(err, webapp.ErrSessionNotFound) {
-			session = nil
+		session, err := ctrl.GetSession(code.WebSessionID)
+		if err != nil {
+			return err
 		}
 
-		var result *webapp.Result
-		if session != nil {
-			result, err = ctrl.InteractionPost(inputFn)
-		} else {
-			result, err = ctrl.EntryPointPost(opts, intent, inputFn)
+		step := session.CurrentStep()
+		if step.FormData == nil {
```

Comment:
Can we assume by convention `FormData` is never nil, since it seems we're switching to use `NewSessionStep` instead of constructing it directly?